src/views/CoCurricularTranscript/index.js

```diff
@@ -86,132 +86,71 @@ export default class Transcript extends Component {
-     Returns: array of Activity components with props Activity and Duration.
-  */
+  // Param: activities - a list of activity objects ("Memberships" as defined in gordon-360-api)
+  // Returns: array of Activity components with props Activity and Sessions.
   groupActivityByCode = activities => {
     let condensedActs = [];
 
     // sort activities by ActivityCode
     while (activities.length > 0) {
       let curAct = activities.shift();
+      let sessions = [];
 
       // keep track of the activity code which will be used to identify all activities of the same
       // code so they can be grouped into one activity component
       let curActCode = curAct.ActivityCode;
 
-      // Pop first session code from array and split into months and years, which are saved as
-      // the initial start and end dates
-      let sess = curAct.SessionCode;
-      let startMon = this.sliceStart(sess);
-      let endMon = this.sliceEnd(sess);
-      let startYear = this.sliceYear(sess),
-        endYear = startYear;
-
       // For each other activity matching curActCode, if it is consecutive to the current end date,
       // save its end date as the new end date, otherwise, add the current start and end dates to
       // the string 'duration' (because the streak is broken) and prepare to start a new streak.
       // Loop assumes activities will be sorted by session and Activity Code.
-      let duration = '';
+      sessions.push(curAct.SessionCode);
       while (activities.length > 0 && activities[0].ActivityCode === curActCode) {
-        sess = activities.shift().SessionCode;
-        let curStartMon = this.sliceStart(sess);
-        let curYear = this.sliceYear(sess);
-        if (this.checkConsecutiveness([endMon, endYear, curStartMon, curYear])) {
-          // a streak of consecutive involvement continues
-          endMon = this.sliceEnd(sess);
-          endYear = this.sliceYear(sess);
-        } else {
-          // a streak has been broken; add its start and end to the string and start new streak
-
-          // don't show the year twice if the months are of the same year
-          if (startYear === endYear) {
-            duration += startMon;
-          } else {
-            duration += startMon + ' ' + startYear;
-          }
-          duration += '-' + endMon + ' ' + endYear + ', ';
-
-          startMon = this.sliceStart(sess);
-          endMon = this.sliceEnd(sess);
-          startYear = endYear = this.sliceYear(sess);
-        }
+        sessions.push(activities[0].SessionCode);
+        activities.shift();
       }
 
-      // Flush the remaining start and end info to duration.
-      // Again, don't show the year twice if the months are of the same year
-      if (startYear === endYear) {
-        duration += startMon;
-      } else {
-        duration += startMon + ' ' + startYear;
-      }
-      duration += '-' + endMon + ' ' + endYear;
+      let sessionsOrdered = sessions.sort(function(a, b) {
```

Comment:
Would it be helpful to add a simple comment that says "sorts sessions from least to most recently"?

Reply:
I don't think so, but if others do, then we could certainly add it. This is one of the default behaviors of sort() available in most JavaScript tutorials, so I don't think it requires clarification.